### PR TITLE
Update bundled HBase to version 1.2.6 and claim support for 1.3.x #567

### PR DESCRIPTION
**Pull request by jerryjch**
Issue #567

Signed-off-by: Jerry He <jinghe@us.ibm.com>

Thank you for contributing to JanusGraph.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [ x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [ x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by jerryjch** [comment]
Good question.  HBase has not moved the 'stable' release up.  It still points to 1.2.6.  
1.3.1 was reported to have failure in large scale IntegrationTest. 
https://www.apache.org/dist/hbase/
I am thinking we claim support for 1.3.1, but keep the 1.2.6 bundled.

File: docs/versions.txt [hunk 1]
```diff
@@ -15,6 +15,6 @@ JanusGraph.
 | JanusGraph | Cassandra | HBase | Bigtable | Elasticsearch | Solr | TinkerPop | Spark | Scala
 | 0.1.0 | 1.2.z, 2.0.z, 2.1.z | 0.98.z, 1.0.z, 1.1.z, 1.2.z | 0.9.z, 1.0.0-preZ | 1.5.z |  5.2.z | 3.2.z | 1.6.z | 2.10.z
 | 0.1.1 | 1.2.z, 2.0.z, 2.1.z | 0.98.z, 1.0.z, 1.1.z, 1.2.z | 0.9.z, 1.0.0-preZ | 1.5.z |  5.2.z | 3.2.z | 1.6.z | 2.10.z
-| 0.2.0 | 1.2.z, 2.0.z, 2.1.z | 0.98.z, 1.0.z, 1.1.z, 1.2.z | 0.9.z, 1.0.0-preZ | 1.z,2.z,5.z,6.z |  5.3-5.5.z, 6.2-6.6.z, 7.z | 3.2.z | 1.6.z | 2.10.z
+| 0.2.0 | 1.2.z, 2.0.z, 2.1.z | 0.98.z, 1.0.z, 1.1.z, 1.2.z, 1.3.z | 0.9.z, 1.0.0-preZ | 1.z,2.z,5.z,6.z |  5.3-5.5.z, 6.2-6.6.z, 7.z | 3.2.z | 1.6.z | 2.10.z
```
Comment (pluradj) [inline docs/versions.txt]: Should we drop 0.98.z and 1.0.z from the matrix? I don't see these anymore on the official mirror http://www-us.apache.org/dist/hbase/

Comment (sjudeng) [reply docs/versions.txt]: @pluradj Would this apply across the chart? For example we have ES 1.x and 2.x even though they're not supported anymore. In my opinion it's okay to advertise compatibility with older versions. It's up to users which versions they choose (or are forced to) use in their environments.

Comment (pluradj) [reply docs/versions.txt]: Fair enough, maybe we just need a disclaimer. Something like: "Although JanusGraph may be compatible with older and no longer supported versions of its dependencies, users are warned that there are possible risks and security exposures with running software that is no longer supported or updated. Please check with the software providers to understand their supported versions. Users are strongly encouraged to use the latest versions of the software."

**Comment by sjudeng** [comment]
I have a Travis job running tests on all 1.1.x versions now to address the question above (see https://travis-ci.org/sjudeng/janusgraph/builds/281648721). Initial results suggest there might be issues with 1.1.10-1.1.12 but 1.1.9 and below are okay. @jerryjch Do you have experience using HBase 1.1.10, 1.1.11 or 1.1.12 with JanusGraph?

**Comment by jerryjch** [comment]
@sjudeng  I don't have experience with the 1.1.10, 1.1.11 or the 1.1.12. I don't think we ever run tests against them after they came out.  Let's me take a quick look to see what is wrong and if it can be fixed.
Otherwise, we will remove them from the version list.
Thanks for running the testing!

**Comment by jerryjch** [comment]
I ran the same testing locally against HBase 1.1.12. On the hbase server log that was generated by the test run:
```
2017-09-30 12:49:07,300 INFO  [main] util.VersionInfo: HBase 1.1.12

2017-09-30 12:49:08,708 ERROR [main] master.HMasterCommandLine: Master exiting
java.lang.RuntimeException: Failed construction of Master: class org.apache.hadoop.hbase.master.HMasterCommandLine$LocalHMaster
        at org.apache.hadoop.hbase.util.JVMClusterUtil.createMasterThread(JVMClusterUtil.java:143)
        at org.apache.hadoop.hbase.LocalHBaseCluster.addMaster(LocalHBaseCluster.java:220)
        at org.apache.hadoop.hbase.LocalHBaseCluster.<init>(LocalHBaseCluster.java:155)
        at org.apache.hadoop.hbase.master.HMasterCommandLine.startMaster(HMasterCommandLine.java:223)
        at org.apache.hadoop.hbase.master.HMasterCommandLine.run(HMasterCommandLine.java:138)
        at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:70)
        at org.apache.hadoop.hbase.util.ServerCommandLine.doMain(ServerCommandLine.java:126)
        at org.apache.hadoop.hbase.master.HMaster.main(HMaster.java:2522)
Caused by: java.lang.NoSuchMethodError: com.google.common.hash.HashFunction.hashString(CharSequence input)(Ljava/lang/CharSequence;)Lcom/google/common/hash/HashCode;
        at org.apache.hadoop.hbase.regionserver.ScannerIdGenerator.<init>(ScannerIdGenerator.java:44)
        at org.apache.hadoop.hbase.regionserver.RSRpcServices.start(RSRpcServices.java:997)
        at org.apache.hadoop.hbase.regionserver.HRegionServer.<init>(HRegionServer.java:596)
        at org.apache.hadoop.hbase.master.HMaster.<init>(HMaster.java:398)
        at org.apache.hadoop.hbase.master.HMasterCommandLine$LocalHMaster.<init>(HMasterCommandLine.java:306)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
```
The Guava method `HashFunction.hashString(CharSequence input)` has been removed since Guava 16.0.  https://google.github.io/guava/releases/15.0/api/docs/com/google/common/hash/HashFunction.html#hashString(java.lang.CharSequence)

The new hbase dependency was introduced by this JIRA: https://issues.apache.org/jira/browse/HBASE-18122

This JIRA happens to be in HBase 1.1.11 and 1.1.12.  But it is not in the released versions of 1.2.x yet.

**Comment by jerryjch** [comment]
HBase 1.x is on older Guava 14.  We had seen incompatibility on the client side previously. But this JIRA introduced incompatibility on the server side.  When running the testing, JanusGraph's Guava 18 is in the hbase server classpath `/root/.m2/repository/com/google/guava/guava/18.0/guava-18.0.jar`. That is why the hbase server could not be started.  Then every test hangs and fails.

**Comment by jerryjch** [comment]
The errors against HBase 1.1.10 look ok:

```
Tests in error:
  HBaseDistributedStoreManagerTest.setUp:47 Â» Runtime java.lang.NullPointerExcep...
  HBaseDistributedStoreManagerTest.tearDown:52 NullPointer
  HBaseIDAuthorityTest>IDAuthorityTest.setUp:129 Â» Runtime java.lang.NullPointer...
  HBaseIDAuthorityTest>IDAuthorityTest.tearDown:163->IDAuthorityTest.close:168 Â» NullPointer
  HBaseIDAuthorityTest>IDAuthorityTest.setUp:129 Â» Runtime java.lang.NullPointer...
  HBaseIDAuthorityTest>IDAuthorityTest.tearDown:163->IDAuthorityTest.close:168 Â» NullPointer

Tests run: 176, Failures: 0, Errors: 6, Skipped: 2
```
I have seen them before.  https://issues.apache.org/jira/browse/HBASE-16732

**Comment by sjudeng** [comment]
Thanks for all doing all the investigation on this question. So ultimately you recommend continuing to advertise full 1.1.x support?

**Comment by jerryjch** [comment]
Yes. We are fine.  HBase server will not be run in the same JVM as JanusGraph.

**Comment by sjudeng** [comment]
Thanks again for running it down.

**Pull request merged**